service_quotas/ec2_limits_test.go

```diff
@@ -353,3 +347,134 @@ func TestStandardInstancesCPUs(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, int64(12), cpus)
 }
+
+func TestAvailableIpsPerSubnetUsageWithError(t *testing.T) {
+	mockClient := &mockEC2Client{
+		err:                     errors.New("some err"),
+		DescribeSubnetsResponse: nil,
+	}
+
+	check := AvailableIpsPerSubnetUsageCheck{mockClient}
+	usage, err := check.Usage()
+
+	assert.Error(t, err)
+	assert.True(t, errors.Is(err, ErrFailedToGetAvailability))
```

Comment:
nit: I think it makes sense to reuse the `ErrFailedToGetUsage` here as well?

Reply:
Since this one uses the ec2 api and the others use the service quotas api, I wanted to create some kind of separation between them with different errors so it's easier to debug?

Reply:
Everything here (ec2 limits) is using only the EC2 API. All checks are also implementing the UsageCheck interface and it makes sense to have a generic error for them.